includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -999,19 +1002,54 @@ private function get_disallowed_attributes_in_node( DOMElement $node, $attr_spec
 		return $attrs_to_remove;
 	}
 
+	/**
+	 * Adapt width and height to conform to the AMP layout.
+	 *
+	 * @param DOMElement $node Node.
+	 */
+	private function adapt_dimensions_to_layout( DOMElement $node ) {
+		$layout = $node->getAttribute( 'layout' );
+
+		if ( ! amp_is_attribute_empty( $layout ) ) {
```

Comment:
This could rather use `$node->hasAttribute('layout')`

Reply:
Note also that we support a `data-amp-layout` attribute which should get converted into `layout` on AMP pages. This is intended to allow authors to define the attribute on HTML pages without adding unrecognized HTML5 attributes.